system/HTTP/IncomingRequest.php

```diff
@@ -342,21 +342,23 @@ public function getJSON(bool $assoc = false, int $depth = 512, int $options = 0)
 	/**
 	 * Get a specific variable from a JSON input stream
 	 *
-	 * @param  string       $index  The variable that you want which can use dot syntax for getting specific values.
-	 * @param  boolean      $assoc  If true, return the result as an associative array.
-	 * @param  integer|null $filter Filter Constant
-	 * @param  mixed        $flags
+	 * @param  string             $index  The variable that you want which can use dot syntax for getting specific values.
+	 * @param  boolean            $assoc  If true, return the result as an associative array.
+	 * @param  integer|null       $filter Filter Constant
+	 * @param  array|integer|null $flags  Option
 	 * @return mixed
 	 */
-	public function getJsonVar(string $index, bool $assoc = false, $filter = null, $flags = null)
+	public function getJsonVar(string $index, bool $assoc = false, ?int $filter = null, $flags = null)
 	{
 		helper('array');
 
 		$data = dot_array_search($index, $this->getJSON(true));
 
 		if (! is_array($data))
 		{
-			$filter = $filter ?? FILTER_DEFAULT;
+			$filter = is_numeric($filter) ? (int) $filter : FILTER_DEFAULT;
```

Comment:
should the typecasting be unnecessary here because you have already typehinted it already as `?int`, so the null coalesce should remain

Reply:
Hm... I agree.